discojs/src/client/decentralized/utils.ts

```diff
@@ -0,0 +1,23 @@
+// Time to wait between network checks in milliseconds.
+const TICK = 100
+
+// Time to wait for the others in milliseconds.
+const MAX_WAIT_PER_ROUND = 10_000
+
+// check if a given boolean condition is true, checks continuously until maxWait time is reached
+export async function pauseUntil (condition: () => boolean): Promise<void> {
```

Comment:
Is there a way to avoid this `pauseUntil` function? Maybe using some TS-oriented ways? I was trying to do that using Promises, or Events. I would be nice to drop this function because I don't really like this way of waiting until a condition is true in TS.

Reply:
I totally agree, that's quite a bad way to do async. I though about it but there I didn't find an easy fix for that.. but there is a way :)

the issue is that data generation (the WS & peer connections) and data consumption (the other funcs) aren't linked. the best way would have to represent theses connections as an async stream/iterable/generator of messages, so that the consumers can await the next message and directly act on the received value. that would also ensure that the server/peers are sending the correct sequence of messages

so I would look into theses [async generators](https://javascript.info/async-iterators-generators#async-generators-finally), or a good old [EventEmitter](https://danilafe.com/blog/typescript_typesafe_events/)

Reply:
thanks, sounds like a good plan. we can make a separate issue for it, if we want to pursue it.